ConsentViewController/Classes/SPConsentManager.swift

```diff
@@ -262,6 +262,19 @@ import UIKit
             campaignType: error.campaignType
         )
     }
+
+    private func selectPrivacyManagerId(fallbackId:String, groupPmId:String?, childPmId:String?) -> String
+    {
+        let hasGroupPmId = groupPmId != nil && groupPmId != ""
+        let hasChildPmId = childPmId != nil && childPmId != ""
+        if hasChildPmId, let childPmId = childPmId {
+            return childPmId
+        }
+        if hasGroupPmId, !hasChildPmId {
```

Comment:
Beautiful! 🤩

Reply:
Nice touch encapsulating the logic to select the pm id in a method 💯